### ISSUE TITLE
Build failure for 0.3.3-rc1 on OpenBSD 6.5

### ISSUE THREAD

**Issue by pwrdwnsys** [jwilm/alacritty#2529]
OpenBSD 6.5 (Release) / amd64. Rust 1.33.0 from OpenBSD packages. Checked out the  v0.3.3-rc1 tag.

```
error[E0425]: cannot find function `lutimes` in module `libc`
   --> /home/pwrdwnsys/.cargo/registry/src/github.com-1ecc6299db9ec823/nix-0.12.1/src/sys/stat.rs:216:15
    |
216 |         libc::lutimes(cstr.as_ptr(), &times[0])
    |               ^^^^^^^
help: a function with a similar name exists
    |
216 |         libc::futimes(cstr.as_ptr(), &times[0])
    |               ^^^^^^^
help: possible candidate is found in another module, you can import it into scope
    |
1   | use sys::stat::lutimes;
    |
```
After finding nix-related problems in issue #2244, I tried those suggestions by bumping `copypasta/Cargo.toml`'s wayland-client to the latest (0.23) and also did a `cargo update`, but build still failed with the same error.

**Comment by chrisduerr** [comment]
I've sent an upstream PR to update the dependencies of smithay-clipboard, so this is currently blocked on https://github.com/Smithay/smithay-clipboard/pull/5.

**Comment by chrisduerr** [comment]
@pwrdwnsys Can you verify if the #2530 PR builds on BSD?

**Comment by pwrdwnsys** [comment]
@chrisduerr Thank you for the rapid response. I checked out #2530 locally, the build is now failing at:
```
error[E0599]: no variant named `EPROTO` found for type `nix::errno::Errno` in the current scope
  --> /home/pwrdwnsys/.cargo/registry/src/github.com-1ecc6299db9ec823/wayland-client-0.23.3/src/native_lib/display.rs:72:40
   |
72 |         if ret == ::nix::errno::Errno::EPROTO as i32 {
   |                   ---------------------^^^^^^
   |                   |
   |                   variant not found in `nix::errno::Errno`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
error: Could not compile `wayland-client`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

It looks like the problem is referenced at https://github.com/rust-windowing/winit/issues/825

**Comment by chrisduerr** [comment]
I've opened a few PRs upstream to update to nix 0.14.1. I'll update my PR and let you know when this has propagated all the way through to Alacritty.

**Comment by chrisduerr** [comment]
@pwrdwnsys would you mind testing again, all old nix deps should be removed now.

**Comment by pwrdwnsys** [comment]
@chrisduerr Thank you very much for your efforts. I ran a new build against #2530, and it made it a lot further - but now I'm getting `LLVM ERROR: out of memory` on my 8GB machine. I have another box with more RAM, I will get OpenBSD setup on it and let you know.

**Comment by kchibisov** [comment]
@pwrdwnsys when do you get `LLVM ERROR`?

**Comment by chrisduerr** [comment]
Rust can take a lot of memory to compile, so this isn't the first time I've heard of it running out of memory and crashing. Usually I'd expect that you can definitely make it work with 8GB if you add some swap and close other programs, but if you have another machine that's definitely an option.

**Comment by pwrdwnsys** [comment]
@kchibisov here is the tail of the build log with the error:
```
...
Compiling font v0.1.0 (/home/pwrdwnsys/building/alacritty/font)
Compiling smithay-client-toolkit v0.4.6
Compiling smithay-client-toolkit v0.6.2
Compiling winit v0.19.1
Compiling smithay-clipboard v0.3.3
Compiling copypasta v0.6.0 (/home/pwrdwnsys/building/alacritty/copypasta)
Compiling glutin v0.21.0
LLVM ERROR: out of memory
error: Could not compile `alacritty`.

Caused by:
process didn't exit successfully: `rustc --edition=2018 --crate-name alacritty alacritty/src/main.rs --color always --crate-type bin --emit=dep-info,link -C opt-level=3 -C lto -C debuginfo=1 --cfg 'feature="default"' -C metadata=1bf88d9c3b1083d0 -C extra-filename=-1bf88d9c3b1083d0 --out-dir /home/pwrdwnsys/building/alacritty/target/release/deps -L dependency=/home/pwrdwnsys/building/alacritty/target/release/deps --extern alacritty_terminal=/home/pwrdwnsys/building/alacritty/target/release/deps/libalacritty_terminal-4faeacf19860b494.rlib --extern clap=/home/pwrdwnsys/building/alacritty/target/release/deps/libclap-18832f3e2c6cf047.rlib --extern crossbeam_channel=/home/pwrdwnsys/building/alacritty/target/release/deps/libcrossbeam_channel-a891ac554a796f8a.rlib --extern env_logger=/home/pwrdwnsys/building/alacritty/target/release/deps/libenv_logger-81f22319b4b5c387.rlib --extern log=/home/pwrdwnsys/building/alacritty/target/release/deps/liblog-665c89cad9801a49.rlib --extern serde_yaml=/home/pwrdwnsys/building/alacritty/target/release/deps/libserde_yaml-1c4d037bb99d6e02.rlib --extern time=/home/pwrdwnsys/building/alacritty/target/release/deps/libtime-a53f22833cc03dc6.rlib --extern xdg=/home/pwrdwnsys/building/alacritty/target/release/deps/libxdg-834529ed1e8831a1.rlib -L native=/home/pwrdwnsys/building/alacritty/target/release/build/libloading-bb4af0b469a76b6d/out -L native=/usr/X11R6/lib -L native=/usr/X11R6/lib` (signal: 6, SIGABRT: process abort signal)
```
It was a clean install of 6.5 with only X and sshd running on the machine, I set it up specifically to test and document my build procedures for a new environment. I'm prepping a VM for a new install now with 16GB RAM and will report back shortly.

**Comment by pwrdwnsys** [comment]
@chrisduerr The build failed again in the same place and with the same message on the 16GB machine. However, I was watching the processes and memory usage with htop - it was only at approx 2.1GB used when it failed.

So I did some digging and found out that LLVM was hitting the maximum datasize limit of the default user login class configuration. Running `ulimit -a` told me I could utilise up to 1.5G, so I modified the default `/etc/login.conf` entry for the `staff` group to the following:
```
staff:\
  :datasize-cur=4096M:\
  :datasize-max=8192M:\
  :maxproc-cur=512:\
  :maxproc-max=1024:\
  :openfiles-cur=4096:\
  :openfiles-max=8192:\
  :stacksize-cur=32M:\
  :ignorenologin:\
  :requirehome@:\
  :tc=default:
```
Then added my user to the staff group with `usermod -G staff MYUSERNAME`, logged out and back in again then re-ran the build. Watching htop I saw total system memory usage hit around 2.6GB just before the build successfully completed.

I'll submit a PR to amend the OpenBSD build instructions shortly in case anyone else should bump into this.
